Sara/tec 294 docs add GitHub app installation and connection workflow

s-santillan opened the pull request:
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR

- Fixes TEC-294. Additionally adds a callout that says SSO must have been set up first (because "usually" SSO is set up after) before performing the connection.
- Updates "integration" with "connection" because "connect" is used in the Semgrep AppSec Platform UI.
- **I am not sure about using double nested tabs**, feedback welcome.

### Static preview

![Snag_106b5b9](https://github.com/user-attachments/assets/e3f7851e-edb9-4148-8244-c39900e09bd9)

jkinsfather commented:
Overall, an accurate and helpful addition.

One of the things I worry about is confusion due to the limitations of tabs and hyperlinks. 

Often, users do not navigate to this page organically. They are sent links by our support and sales teams. Since we can't send someone a hyperlink of a tab, the process for the non-GitHub SSO user would be the following:
1. They are sent and open a hyperlink to the "Connect to cloud-hosted orgs" header. 
2. They click on the GitHub cloud tab. 
3. They are automatically shown the "Default GitHub Cloud" tab. Now they need to either:
    3a. Read and correctly follow written instructions given by our team to switch to the `...with non-GitHub SSO` tab. 
    3b. Notice and infer that ' ... with non-GitHub SSO' refers to the method by which they logged in to Semgrep and NOT the log-in method to say, their GitHub Cloud Org. 
    
To avoid confusion, would it make sense to do one or both of the following?:
1. Add some kind of anchor under the tabs/sub-tabs so we can link directly to the tabs without needing to deliver additional instructions to the customer for navigation? 
2. Add an info box under the `Default GitHub Cloud` sub-tab explaining that `This is for Semgrep orgs set up with GitHub SSO for authentication. If your org is set up with non-GitHub SSO, follow the instructions in the Non-GitHub SSO tab.`

s-santillan commented:
To avoid confusion, would it make sense to do one or both of the following?:

> Add some kind of anchor under the tabs/sub-tabs so we can link directly to the tabs without needing to deliver additional instructions to the customer for navigation?

Let's say the user needs the instructions for Bitbucket and we add a header `### Bitbucket Cloud` nested in the tab - there's unfortunately no capability for the tab to "select itself" or to "show itself" - you'd still have to click on the tab, then the TOC link on the right hand side will work. This still isn't ideal as it requires the user to press the tab. 

I see your point about the need to link directly, or with the least amount of steps, especially considering how crucial deployment is. In this case it may be best **to break out each SCM into its own doc**. (cc @khorne3 for feedback)

For now, I can:

1. Remove the subnesting (as it will require the user to Click **Connect to cloud-hosted orgs > Click GitHub Cloud > Click GitHub with non-GitHub SSO** which is a frankly painful sentence to type out.
2. Follow your second recommendation: _Add an info box under the Default GitHub Cloud sub-tab explaining that This is for Semgrep orgs set up with GitHub SSO for authentication. If your org is set up with non-GitHub SSO, follow the instructions in the Non-GitHub SSO tab._

s-santillan commented:
@jkinsfather @khorne3 I updated the PR, addressing the comments above: https://deploy-preview-1989--semgrep-docs-prod.netlify.app/docs/deployment/connect-scm#github-cloud-with-non-github-sso

khorne3 reviewed (approved):
@s-santillan I agree with the need for individual SCM pages, both here and elsewhere. Perhaps our next big non-feature related project?

jkinsfather commented on docs/deployment/connect-scm.md:
```diff
@@ -67,34 +67,46 @@ If you opted to scan a GitHub or GitLab repository when you initially signed in,
-- Getting Assistant recommendations grants Semgrep **code access**.
-- **Leave PR comments** refers to Semgrep's capability to post findings to developers in PRs.
-:::
+These instructions are for orgs that set up a non-GitHub SSO **before** performing any of the steps in this document.
```
Use of orgs is ambiguous. Perhaps re-phrase to mirror the GitHub SSO instructions? "These steps are for users that sign in to Semgrep through a non-GitHub SSO provider"

s-santillan replied:
done!
